### PR TITLE
fix: Windows commands

### PR DESCRIPTION
**Pull request by xtuc**
Port of #147 

Closes https://github.com/cloudflare/wrangler/issues/153

### CODE REVIEW

File: tests/build.rs [hunk 1]
```diff
@@ -43,13 +43,21 @@ fn it_builds_with_webpack_single_js() {
 
 fn cleanup(fixture: &str) {
     let path = fixture_path(fixture);
-    assert!(path.exists());
-    fs::remove_dir_all(path.clone()).unwrap();
+    assert!(path.exists(), format!("{:?} does not exists", path));
```
Comment (steveklabnik) [inline tests/build.rs]:
```suggestion
    assert!(path.exists(), format!("{:?} does not exist", path));
```

**Comment by steveklabnik** [comment]
Tests now pass for me locally:

```text
~\src\wrangler [master ≡]> cargo test
   Compiling escargot v0.4.0
   Compiling predicates-core v1.0.0
   Compiling treeline v0.1.0
   Compiling difference v2.0.0
   Compiling fs_extra v1.1.0
   Compiling wrangler v0.1.1 (C:\Users\Steve Klabnik\src\wrangler)
   Compiling predicates-tree v1.0.0
   Compiling predicates v1.0.1
   Compiling assert_cmd v0.11.1
    Finished dev [unoptimized + debuginfo] target(s) in 7.18s
     Running target\debug\deps\wrangler-2cc44cf45d8cc57e.exe

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target\debug\deps\build-7746098309fb5477.exe

running 1 test
test it_builds_with_webpack_single_js ... FAILED

failures:

---- it_builds_with_webpack_single_js stdout ----
dir: "C:\\Users\\STEVEK~1\\AppData\\Local\\Temp\\webpack_simple_js"
thread 'it_builds_with_webpack_single_js' panicked at 'Unexpected failure.
code-101
stderr=```thread 'main' panicked at 'could not run `npm install`: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }', src\l
ibcore\result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
command=`"C:\\Users\\Steve Klabnik\\src\\wrangler\\target\\debug\\wrangler.exe" "build"`
code=101
stdout=``````
stderr=```thread 'main' panicked at 'could not run `npm install`: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }', src\l
ibcore\result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
', C:\Users\Steve Klabnik\.cargo\registry\src\github.com-1ecc6299db9ec823\assert_cmd-0.11.1\src\assert.rs:154:13
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.


failures:
    it_builds_with_webpack_single_js

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--test build'
~\src\wrangler [master ≡]> git checkout fix-windows-invoke
Switched to a new branch 'fix-windows-invoke'
Branch 'fix-windows-invoke' set up to track remote branch 'fix-windows-invoke' from 'upstream'.
~\src\wrangler [fix-windows-invoke ≡]> cargo test
    Updating git repository `https://github.com/xtuc/binary-install.git`
   Compiling wrangler v0.1.1 (C:\Users\Steve Klabnik\src\wrangler)
   Compiling binary-install v0.0.2 (https://github.com/xtuc/binary-install.git?branch=feat-download-artifact#661d8860)
    Finished dev [unoptimized + debuginfo] target(s) in 7.57s
     Running target\debug\deps\wrangler-9b190d0fbb49b2d0.exe

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target\debug\deps\build-8f3b3b3cf28fb968.exe

running 1 test
test it_builds_with_webpack_single_js ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target\debug\deps\generate-ecc9d44185f5f20a.exe

running 2 tests
test it_generates_with_defaults ... ok
test it_generates_with_arguments ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target\debug\deps\help-615c9c8a7cf40980.exe

running 3 tests
test it_works ... ok
test it_has_a_help_flag ... ok
test it_fails_on_bad_command ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

~\src\wrangler [fix-windows-invoke ≡]>
```

File: src/wranglerjs/mod.rs [hunk 2]
```diff
@@ -183,6 +183,23 @@ pub fn run_npm_install(dir: PathBuf) -> Result<(), failure::Error> {
     }
 }
 
+// build a Command for npm
+//
+// Here's the deal: on Windows, `npm` isn't a binary, it's a shell script.
+// This means that we can't invoke it via `Command` directly on Windows,
+// we need to invoke `cmd /C npm`, to run it within the cmd environment.
+fn build_npm_command() -> Command {
+    if install::target::WINDOWS {
```
Comment (steveklabnik) [inline src/wranglerjs/mod.rs]: why choose `install::target::WINDOWS` over `cfg!(target_os = "windows")`?

File: tests/build.rs [hunk 3]
```diff
@@ -43,13 +43,21 @@ fn it_builds_with_webpack_single_js() {
...
+
+    // Workaround https://github.com/rust-lang/rust/issues/29497
+    if cfg!(target_os = "windows") {
+        let mut command = Command::new("cmd");
+        command.arg("rmdir");
+        command.arg("/s");
+        command.arg(path.clone());
```
Comment (steveklabnik) [inline tests/build.rs]: is this clone actually needed?

File: tests/build.rs [hunk 4]
```diff
@@ -43,13 +43,21 @@ fn it_builds_with_webpack_single_js() {
...
+    } else {
+        fs::remove_dir_all(path.clone()).unwrap();
```
Comment (steveklabnik) [inline tests/build.rs]: is this clone actually needed?

File: src/wranglerjs/mod.rs (hunk 2 above)
Comment (ashleygwilliams) [reply src/wranglerjs/mod.rs]: this would be the one thing i'd change but we can do it in another PR i think

Comment (xtuc) [reply src/wranglerjs/mod.rs]: I used it for consistency, but we can change it.

**Pull request merged**